### DISCUSSION TITLE
[Art competition] Invariant

### DISCUSSION THREAD

**Discussion by hyerinshelly** [prosyslab-classroom/cs424-program-reasoning#249] [Art competition]
![DALL·E 2022-11-29 17 57 20 - One twinkling star that shines brightly on the girl's head in the vast night sky, while other numerous starlights are shining darker](https://user-images.githubusercontent.com/47997946/204484468-d08aec11-9aec-43ec-a338-bfb79236e559.png)

## Short Description
For me, [*invariant*] means [*one twinkling star that shines brightly on the girl's head in the vast night sky, while other numerous starlights are shining darker*]".

## Long Description (up to 3 sentences)
Invariant of a program is the property that represents the program during its whole execution, even though we cannot know when it will terminate. In terms of this, I think finding the invariant of every program is similar to the goal of every human's life, while everyone never knows when life will end. Every life is shiny in its own way but we may not notice it, so finding one star, that looks just like you and shines exactly on you, among countless stars and knowing how it is twinkling are what we need to do during our whole lifetime.

**Comment by KihongHeo** [comment]
Beautiful invariant!
